extensions/interactions/PencilCodeEditor/directives/oppia-interactive-pencil-code-editor.component.ts

```diff
@@ -0,0 +1,223 @@
+  }
+
+  ngOnInit(): void {
+    this.directiveSubscriptions.add(
+      this.playerPositionService.onNewCardAvailable.subscribe(
+        () => {
+          this.interactionIsActive = false;
+          this.pce.hideMiddleButton();
+          this.pce.hideToggleButton();
+          this.pce.setReadOnly();
+        }
+      )
+    );
+
+    this.iframeDiv = this.elementRef.nativeElement.querySelectorAll(
+      '.pencil-code-editor-iframe')[0];
+    this.pce = new PencilCodeEmbed(this.iframeDiv);
+    this.interactionIsActive = (this.lastAnswer === null);
+
+    const { initialCode } = (
+      this.interactionAttributesExtractorService.getValuesFromAttributes(
+        'PencilCodeEditor',
+        this._getAttributes()
+      ) as PencilCodeEditorCustomizationArgs);
+    this.someInitialCode = (
+      this.interactionIsActive ? initialCode.value : this.lastAnswer.code);
+
+    this.pce.beginLoad(this.someInitialCode);
+    this.pce.on('load', () => {
+      // Hides the error console at the bottom right, and prevents it
+      // from showing up even if the code has an error. Also, hides the
+      // turtle, and redefines say() to also write the text on the
+      // screen.
+      this.pce.setupScript([{
+        code: [
+          'window.onerror = function() {',
+          '  return true;',
+          '};',
+          'debug.hide();',
+          'window.removeEventListener("error", debug)',
+          '',
+          'ht();',
+          '',
+          'oldsay = window.say',
+          'say = function(x) {',
+          '  write(x);',
+          '  oldsay(x);',
+          '};'
+        ].join('\n'),
+        type: 'text/javascript'
+      }]);
+
+      this.pce.showEditor();
+      this.pce.hideToggleButton();
+      if (this.interactionIsActive) {
+        this.pce.setEditable();
+      } else {
+        this.pce.hideMiddleButton();
+        this.pce.setReadOnly();
+      }
+
+      // Pencil Code automatically takes the focus on load, so we clear
+      // it.
+      this.focusManagerService.clearFocus();
+    });
+
+    let errorIsHappening = false;
+    let hasSubmittedAnswer = false;
+
+    this.pce.on('startExecute', () => {
+      hasSubmittedAnswer = false;
+    });
+
+    this.pce.on('execute', () => {
+      if (errorIsHappening || hasSubmittedAnswer) {
+        return;
+      }
+      // The first argument in the method below gets executed in the
+      // pencilcode output-frame iframe context. The code input by the
+      // user is sanitized by pencilcode so there is no security
+      // issue in this case.
+      this.pce.eval(
+        'document.body.innerHTML', // disable-bad-pattern-check
+        (pencilCodeHtml) => {
+          let normalizedCode = this.getNormalizedCode();
+
+          // Get all the divs, and extract their textual content.
+          let temp = document.createElement('div');
+          // eslint-disable-next-line oppia/no-inner-html
+          temp.innerHTML = pencilCodeHtml;
```

Comment:
Add a comment explaining why disabling the lint check was the last resort taken.

Reply:
done